redirect_cache.c

```diff
@@ -33,62 +33,52 @@
-	}
-	if (redirect_info->redirect_host) {
-		mnd_pefree(redirect_info->redirect_host, 1);
-		redirect_info->redirect_host = NULL;
-	}
-	if (redirect_info) {
+
+    if (redirect_info != NULL) {
+
+        if (redirect_info->redirect_user) {
+            mnd_pefree(redirect_info->redirect_user, 1);
+            redirect_info->redirect_user = NULL;
+        }
+        if (redirect_info->redirect_host) {
+            mnd_pefree(redirect_info->redirect_host, 1);
+            redirect_info->redirect_host = NULL;
+        }
+
 		mnd_pefree(redirect_info, 1);
 		redirect_info = NULL;
-	}
-}
-/* }}} */
-
-/* {{{ mysqlnd_azure_get_is_using_redirect */
-MYSQLND_AZURE_CONN_DATA** mysqlnd_azure_get_is_using_redirect(const MYSQLND_CONN_DATA *conn)
-{
-	MYSQLND_AZURE_CONN_DATA** props;
-	props = (MYSQLND_AZURE_CONN_DATA**)mysqlnd_plugin_get_plugin_connection_data_data(conn, mysqlnd_azure_plugin_id);
-	if (!props || !(*props)) {
-		*props = mnd_pecalloc(1, sizeof(MYSQLND_AZURE_CONN_DATA), conn->persistent);
-		(*props)->is_using_redirect = 0;
-	}
-	return props;
-}
-/* }}} */
+      
+    }
 
-/* {{{ mysqlnd_azure_set_is_using_redirect */
-MYSQLND_AZURE_CONN_DATA** mysqlnd_azure_set_is_using_redirect(MYSQLND_CONN_DATA *conn, zend_bool is_using_redirect)
-{
-	MYSQLND_AZURE_CONN_DATA** props;
-	props = (MYSQLND_AZURE_CONN_DATA**)mysqlnd_plugin_get_plugin_connection_data_data(conn, mysqlnd_azure_plugin_id);
-	if (!props || !(*props)) {
-		*props = mnd_pecalloc(1, sizeof(MYSQLND_AZURE_CONN_DATA), conn->persistent);
-		(*props)->is_using_redirect = 0;
-	}
-	(*props)->is_using_redirect = is_using_redirect;
-	return props;
 }
 /* }}} */
 
 /* {{{ mysqlnd_azure_add_redirect_cache */
-enum_func_status mysqlnd_azure_add_redirect_cache(const MYSQLND_CONN_DATA* conn, const char* user, const char* host, int port, const char* redirect_user, const char* redirect_host, int redirect_port)
+enum_func_status mysqlnd_azure_add_redirect_cache(const char* user, const char* host, int port, const char* redirect_user, const char* redirect_host, int redirect_port)
 {
 	if (MYSQLND_AZURE_G(redirectCache) == NULL) {
 		MYSQLND_AZURE_G(redirectCache) = mnd_pemalloc(sizeof(HashTable), 1);
+        if(MYSQLND_AZURE_G(redirectCache) == NULL) {
+            return FAIL;
+        }
 		zend_hash_init(MYSQLND_AZURE_G(redirectCache), 0, NULL, mysqlnd_azure_redirect_info_dtor, 1);
 	}
 
-	char *key = NULL;
+	char* key = NULL;
 	mnd_sprintf(&key, MAX_REDIRECT_HOST_LEN+ MAX_REDIRECT_USER_LEN+8, "%s_%s_%d", user, host, port);
-
-	MYSQLND_AZURE_REDIRECT_INFO* redirect_info = pemalloc(sizeof(MYSQLND_AZURE_REDIRECT_INFO), conn->persistent);
-	redirect_info->redirect_user = mnd_pestrndup(redirect_user, strlen(redirect_user), conn->persistent);
-	redirect_info->redirect_host = mnd_pestrndup(redirect_host, strlen(redirect_host), conn->persistent);
+    if(!key) {
+        return FAIL;
+    }
+
+	MYSQLND_AZURE_REDIRECT_INFO* redirect_info = pemalloc(sizeof(MYSQLND_AZURE_REDIRECT_INFO), 1);
```

Comment:
code format

Reply:
About the format, I think it maybe editor related. It looks with right indent with my editor...

Reply:
That's tabs vs. spaces.

Reply:
Thanks a lot for your information. Yet I choose not to adjust the formatting this time, thanks